tech-specs/src/explore/design/ibc.md

```diff
@@ -0,0 +1,408 @@
+                                    //   - Use `ibc-rs::ics04_connection::handler::verify::verify_channel_proofs()`
+                                }
+                                None => {
+                                    // "ChanCloseInit"
+                                    continue;
+                                }
+                            }
+                        }
+                        StateChange::Updated => {
+                            // Verify the proof to check the counterpart ledger's state is expected
+                            //   - The state can be inferred from the own channel state change
+                            //   - Use `ibc-rs::ics04_connection::handler::verify::verify_channel_proofs()`
+                        }
+                        _ => return Err(Error::InvalidStateChange("Invalid state change happened")),
+                    }
+                }
+
+                "nextSequenceSend" => {
+                    // "SendPacket"
+                    let packet = get_packet(key, data_from_tx)?;
+
+                    // Assert that the packet metadata matches the channel and connection information
+                    //   - the port is owend
+                    //   - the channel exists
+                    //   - the counterparty information is valid
+                    //   - the connection exists
+
+                    // Assert that the connection and channel are open
+
+                    // Assert that the packet sequence is the next sequence that the channel expects
+
+                    // Assert that the timeout height and timestamp have not passed on the destination ledger
+
+                    // Assert that the commitment has been stored
+                }
+
+                "nextSequenceRecv" => {
+                    // "RecvPacket"
+                    let packet = get_packet(key, data_from_tx)?;
+                    let proofs = get_proofs(key, data_from_tx)?;
+
+                    // Assert that the packet metadata matches the channel and connection information
+
+                    // Assert that the connection and channel are open
+
+                    // Assert that the packet sequence is the next sequence that the channel expects (Ordered channel)
+
+                    // Assert that the timeout height and timestamp have not passed on the destination ledger
+
+                    // Assert that the receipt and acknowledgement have been stored
+
+                    // Verify the proofs that the counterpart ledger has stored the commitment
+                    //   - Use `ibc-rs::ics04_connection::handler::verify::verify_packet_recv_proofs()`
+                }
+
+                "nextSequenceAck" => {
+                    // "Acknowledgement"
+                    let packet = get_packet(key, data_from_tx)?;
+                    let proofs = get_proofs(key, data_from_tx)?;
+
```

Comment:
is there a function in `ibc-rs` we can call here?

Reply:
`ChannelReader`'s functions and [`verify_packet_acknowledgement_proofs()`](https://github.com/informalsystems/ibc-rs/blob/5cf3b6790c45539c5aaadeef6e1af1f51a5f437f/modules/src/ics04_channel/handler/verify.rs#L101) are called